pkg/statsd/statsd_defaults_and_params.go

```diff
@@ -32,6 +35,43 @@ var DefaultTags = gostatsd.Tags{}
 // DefaultInternalTags is the default list of additional tags on internal metrics
 var DefaultInternalTags = gostatsd.Tags{}
 
+// DefaultCloudProviderCacheValues contains the default cache values for each cloud provider.
+var DefaultCloudProviderCacheValues = map[string]CacheOptions{
+	k8s.ProviderName: {
+		// Basically the CacheRefreshPeriod is the granularity at which we care about batching data.
+		// If 5 requests come in inside 50 milliseconds, and this is set to 50 milliseconds, we only need to look
+		// up the informer once, and do the regexes once. This saves us resources at the cost of data precision.
+		CacheRefreshPeriod:        50 * time.Millisecond,
```

Comment:
I think we should just leave the defaults for now and refactor the things to avoid this problem all together. I'm working on that refactoring in a separate branch to merge later.

Reply:
I disagree. The purpose of these cache overrides is that we can use what's here indefinitely if we absolutely have to. If we use the cache values of the AWS one then if a pod leaves and arrives again with the same IP quickly then we'll cache metrics for far too long.

I don't see any harm in splitting this, basically.

Reply:
ok 👍